tests/test_cases/test_cocotb/test_deprecated.py

```diff
@@ -0,0 +1,23 @@
+import cocotb
+import warnings
+
+
+async def catch_deprecation(coro):
+    with warnings.catch_warnings(record=True) as warn:
+        # Cause all warnings to always be triggered.
+        warnings.simplefilter("always")
+        res = await coro
+        assert len(warn) == 1
+        warn = warn[0]
+        assert issubclass(warn.category, DeprecationWarning)
+        return res, warn
+
+
+@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith("icarus"))
+async def test_unicode_handle_assignment_deprecated(dut):
+    async def write_str():
+        dut.string_input_port <= "Bad idea"
+
+    result, warn = await catch_deprecation(write_str())
```

Comment:
A context manager would be easier here
```suggestion
from contextlib import contextmanager
@contextmanager
def assert_deprecated():
    warn = []
    try:
        with warnings.catch_warnings(record=True) as warn:
            # Cause all warnings to always be triggered.
            warnings.simplefilter("always")
            yield warn  # note: not a cocotb yield, but a contextlib one!
    finally:
        assert len(warn) == 1
        warn = warn[0]
        assert issubclass(warn.category, DeprecationWarning)

@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith("icarus"))
async def test_unicode_handle_assignment_deprecated(dut):
    with assert_deprecated() as w:
        dut.string_input_port <= "Bad idea"
    assert "bytes" in w[0].message
```

Reply:
Since we can't test this with Icarus, I will change my other PR to include this so we have a reasonable good idea it works.

Reply:
No need to change that PR, but once I've merged it you can update the test from that PR to use this function too.